overlord/fdestate/fdemgr.go

```diff
@@ -132,19 +132,39 @@ func (m *FDEManager) ReloadModeenv() error {
 	return initModeFromModeenv(m)
 }
 
+type unlockedStateManager struct {
+	unlocker boot.Unlocker
+	m        *FDEManager
+}
+
+func (m *unlockedStateManager) Update(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error {
+	if m.unlocker != nil {
```

Comment:
so now we unlock to run reseal below, then lock/unlock to update, then lock/unlock again to get, then relock? can we avoid or coalesce some of these?

Reply:
We would have to bring the unlocker down. I think what we need to do use unlock while we calculate the profile. This is what takes time. Then we do not have to relock to Get/Update

Reply:
That sounds good. I will try that.

Reply:
I have changed to do the unlocking only when calculating the profile. And put the unlocker into the FDEStateManager abstraction.